src/structures/buffers.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::tss2_esys::{TPM2_MAX_DIGEST_BUFFER, TPM2_MAX_NV_BUFFER_SIZE};
+use crate::{Error, Result, WrapperErrorKind};
+use log::error;
+use std::convert::TryFrom;
+use std::ops::Deref;
+
+#[allow(unused_macros)]
+macro_rules! buffer_type {
+    ($native_type:ident,$MAX:expr,$tss_type:ident) => {
+        use crate::tss2_esys::$tss_type;
+
+        #[derive(Debug, Clone, PartialEq, Eq, Default)]
+        pub struct $native_type(Vec<u8>);
+        impl $native_type {
+            const MAX_SIZE: usize = $MAX;
+
+            pub fn value(&self) -> &[u8] {
+                &self.0
+            }
+        }
+
+        impl Deref for $native_type {
+            type Target = Vec<u8>;
+            fn deref(&self) -> &Self::Target {
+                &self.0
+            }
+        }
+
+        impl TryFrom<Vec<u8>> for $native_type {
+            type Error = Error;
+
+            fn try_from(bytes: Vec<u8>) -> Result<Self> {
```

Comment:
I wish there was some way we could make a try_from from a slice. But unfortunatly I do not think it is possible.